### PR TITLE
fix: Update siwe parse to v2.1.0

### PR DESCRIPTION
**Pull request by tommasini**
## **Description**
More info [here](https://github.com/MetaMask/MetaMask-planning/issues/2278)

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by NEllusion — DISMISSED** [review]

**Review by Cal-L — COMMENTED** [review]
Fix dedupe and e2e test failures

File: package.json [hunk 1]
```diff
@@ -121,8 +121,9 @@
     "axios": "1.6.0",
     "react-native-ble-plx": "3.1.2",
     "**/babel-runtime/regenerator-runtime": "^0.13.8",
-    "@metamask/transaction-controller/@metamask/controller-utils": "^5.0.0"
-
+    "@metamask/transaction-controller/@metamask/controller-utils": "^5.0.0",
+    "@spruceid/siwe-parser": "2.1.0",
+    "apg-js": "4.2.0"
```
Comment (digiwand) [inline package.json:126]: Q: why is the apg-js package needed here?

Comment (tommasini) [reply package.json:126]: The mobile app was breaking with only the version 2.1.0 of siwe-parser
image

It seems to be a known issue, the proposed solution was found [here](https://github.com/spruceid/siwe/issues/165#issuecomment-1686341772)

**Comment by tommasini** [comment]
@SocketSecurity ignore-all
After talking with @NicholasEllul, the risks discovered by socket security are acceptable

File: package.json (hunk 1 above)
Comment (digiwand) [reply package.json:126]: interesting bug. Thanks for sharing! I'm good with merging this in for now to get the updated siwe-parser in. 

I also [proposed updating apg-js within the spruceid/siwe repo](https://github.com/spruceid/siwe/pull/199) with the information you provided. If merged we could go back to remove this dependency

**Review by Cal-L — APPROVED** [review]
LGTM

**Review by digiwand — APPROVED** [review]
looks good to merge! 

I created a [PR](https://github.com/spruceid/siwe/pull/199) to update the `apg-js` npm package in the spruceid/siwe repo based on https://github.com/MetaMask/metamask-mobile/pull/9247#discussion_r1569578334. If it's merged, we could remove the library from package.json

**Comment by digiwand** [comment]
The PR mentioned above has been merged and released to `@spruceid/siwe-parser@2.1.2`. We could update the siwe-parser package and remove the `apg-js` dependency

--- 

something I was thinking is we might even be able to revert this PR. The reason is because mobile doesn't support SIWE confirmation pages yet and I don't see any use-cases for the `siwe-parser` in the repo 

Mobile ticket to support SIWE: https://github.com/MetaMask/MetaMask-planning/issues/1128

cc: @tommasini @NicholasEllul

**Pull request merged**